sdk/core/azure-core/src/main/java/com/azure/core/util/Context.java

```diff
@@ -23,6 +24,9 @@
 public class Context {
     private final ClientLogger logger = new ClientLogger(Context.class);
 
+    /**Key used to override headers in HttpRequest. The Value for this key should be {@link HttpHeaders}.*/
+    public static String AZURE_REQUEST_HTTP_HEADERS_KEY = "azure-request-override-http-headers-key";
```

Comment:
Does this constant make more sense in `Context` or the policy which uses it? Recently I made changes to `UserAgentPolicy` to look for `Context` values and I put the constants in the policy itself.

@srnagar @JonathanGiles Thoughts on location?

Reply:
I thought from end user point of view, Since in public API we are asking end user to add context.addData(key , value ) , Use will find the Key in Context class. 
But I can change if  everyone suggest another location.

Reply:
I think Context works for now, if anything better can be thought of we can circle back on this.